data/scripts/update_ip_country_data.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# COPYRIGHT (C) 2024 Nicotine+ Contributors
+#
+# GNU GENERAL PUBLIC LICENSE
+#    Version 3, 29 June 2007
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import csv
+import io
+import os
+import urllib.request
+import zipfile
+
+
+def update_ip_country_data():
+    """Download IP2Location country data and convert it for use in Python."""
+
+    country_data_url = "https://download.ip2location.com/lite/IP2LOCATION-LITE-DB1.CSV.ZIP"
+    file_path, _headers = urllib.request.urlretrieve(country_data_url)
+    base_path = os.path.normpath(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", ".."))
+    max_ipv4_range = 4294967295
+
+    country_codes = set()
+    ip_ranges = []
+    country_values = []
+
+    with zipfile.ZipFile(file_path, "r") as zip_file_handle:
+        license_file_name = zip_file_handle.namelist()[0]
+        csv_file_name = zip_file_handle.namelist()[-1]
+
+        with zip_file_handle.open(license_file_name) as license_file_handle:
+            if b"Creative Commons Attribution-ShareAlike 4.0 International" not in license_file_handle.read():
+                raise ValueError("Invalid country data license")
+
+        with zip_file_handle.open(csv_file_name) as csv_file_handle:
+            for row in csv.reader(io.TextIOWrapper(csv_file_handle)):
+                _address_from, address_to, country_code, _country_name = list(row)
+                address_to = int(address_to)
+
+                if (len(country_code) != 2 or not country_code.isupper()) and country_code != "-":
+                    raise ValueError("Invalid country code")
+
+                if address_to < 0 or address_to > max_ipv4_range:
+                    raise ValueError("Invalid IP address")
+
+                country_codes.add(country_code)
+                ip_ranges.append(address_to)
+                country_values.append(country_code)
+
+    output_ip_ranges = bytearray(b"""# IP2Location LITE (c) 2001-2024 Hexasoft Development Sdn. Bhd.
```

Comment:
Suggest automate the insertion of the copyright year text from the source license file.